examples/vision/randaugment.py

```diff
@@ -0,0 +1,296 @@
+"""
+Title: RandAugment for Image Classification for Improved Robustness
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/03/13
+Last modified: 2021/03/13
+Description: Applying RandAugment augmentation for training an image classification model with improved robustness.
+"""
+"""
+Data augmentation is a very useful technique that helps to improve the translational
+invariance of convolutional neural networks (CNN). RandAugment is a stochastic data
+augmentation routine for vision data and was proposed in [RandAugment: Practical
+automated data augmentation with a reduced search
+space](https://arxiv.org/abs/1909.13719). It is composed of strong augmentation
+transforms like color jitters, Gaussian blurs, saturations, etc. along with more
+traditional augmentation transforms such as random crops.
+
+RandAugment has two parameters:
+* `n` that denotes the number of randomly selected augmentation transforms to apply
+sequentially
+* `m` strength of all the augmentation transforms
+
+These parameters are tuned for a given dataset and a network architecture. The authors of
+RandAugment also provide pseudocode of RandAugment in the original paper (Figure 2):
+
+![](https://i.ibb.co/Df6Ynxd/image.png)
+
+Recently, it has been a key component of works like [Noisy Student Training](https://arxiv.org/abs/1911.04252) and
+[Unsupervised Data Augmentation for Consistency Training](https://arxiv.org/abs/1904.12848). It has been also central to the
+success of [EfficientNets](https://arxiv.org/abs/1905.11946).
+
+This example requires TensorFlow 2.4 or higher, as well as
+[`imgaug`](https://imgaug.readthedocs.io/), which can be installed using the following
+command:
+"""
+
+"""shell
+pip install -U -q imgaug
+"""
+
+"""
+## Setup
+"""
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from tensorflow.keras import layers
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+from imgaug import augmenters as iaa
+import imgaug as ia
+
+"""
+## Load the CIFAR10 dataset
+
+For this example, we will be using the [CIFAR10
+dataset](https://www.cs.toronto.edu/~kriz/cifar.html).
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
+print(f"Total training examples: {len(x_train)}")
+print(f"Total test examples: {len(x_test)}")
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 128
+EPOCHS = 10
+
+"""
+## Initialize `RandAugment` object
+
+Now, we will initialize a `RandAugment` object from the `imgaug.augmenters` module with
+the parameters suggested by the RandAugment authors.
+"""
+
+rand_aug = iaa.RandAugment(n=3, m=7)
+
+
+def augment(images):
+    # Input to `augment()` is a TensorFlow tensor which
+    # is not supported by `imgaug`. This is why we first
+    # convert it to its `numpy` variant.
+    return rand_aug(images=images.numpy())
```

Comment:
You could just `lambda` it inside the `py_function`, no?

Reply:
Yes, exactly. But I wanted to explain the part on `image.numpy()`.